module/documents/actor/actor.mjs

```diff
@@ -1574,171 +1579,140 @@ export default class Actor5e extends SystemDocumentMixin(Actor) {
-   * @param {object} options      Options which configure how ability tests are rolled
-   * @returns {Promise<D20Roll>}  A Promise which resolves to the created Roll instance
+   * Roll an Ability Test.
+   * @param {AbilityRollProcessConfiguration} config  Configuration information for the roll.
+   * @param {BasicRollDialogConfiguration} dialog     Configuration for the roll dialog.
+   * @param {BasicRollMessageConfiguration} message   Configuration for the roll message.
+   * @returns {Promise<D20Roll[]|null>}               A Promise which resolves to the created Roll instance.
    */
-  async rollAbilityTest(abilityId, options={}) {
-    const label = CONFIG.DND5E.abilities[abilityId]?.label ?? "";
-    const abl = this.system.abilities[abilityId];
-    const globalBonuses = this.system.bonuses?.abilities ?? {};
-    const parts = [];
-    const data = this.getRollData();
-
-    // Add ability modifier
-    parts.push("@mod");
-    data.mod = abl?.mod ?? 0;
-
-    // Include proficiency bonus
-    if ( abl?.checkProf.hasProficiency ) {
-      parts.push("@prof");
-      data.prof = abl.checkProf.term;
-    }
-
-    // Add ability-specific check bonus
-    if ( abl?.bonuses?.check ) {
-      const checkBonusKey = `${abilityId}CheckBonus`;
-      parts.push(`@${checkBonusKey}`);
-      data[checkBonusKey] = Roll.replaceFormulaData(abl.bonuses.check, data);
-    }
-
-    // Add global actor bonus
-    if ( globalBonuses.check ) {
-      parts.push("@checkBonus");
-      data.checkBonus = Roll.replaceFormulaData(globalBonuses.check, data);
-    }
-
-    // Add exhaustion reduction
-    this.addRollExhaustion(parts, data);
-
-    // Roll and return
-    const flavor = game.i18n.format("DND5E.AbilityPromptTitle", {ability: label});
-    const rollData = foundry.utils.mergeObject({
-      data,
-      title: `${flavor}: ${this.name}`,
-      flavor,
-      halflingLucky: this.getFlag("dnd5e", "halflingLucky"),
-      messageData: {
-        speaker: options.speaker || ChatMessage.getSpeaker({actor: this}),
-        "flags.dnd5e.roll": {type: "ability", abilityId }
-      }
-    }, options);
-    rollData.parts = parts.concat(options.parts ?? []);
-
-    /**
-     * A hook event that fires before an ability test is rolled for an Actor.
-     * @function dnd5e.preRollAbilityTest
-     * @memberof hookEvents
-     * @param {Actor5e} actor                Actor for which the ability test is being rolled.
-     * @param {D20RollConfiguration} config  Configuration data for the pending roll.
-     * @param {string} abilityId             ID of the ability being rolled as defined in `DND5E.abilities`.
-     * @returns {boolean}                    Explicitly return `false` to prevent ability test from being rolled.
-     */
-    if ( Hooks.call("dnd5e.preRollAbilityTest", this, rollData, abilityId) === false ) return;
-
-    const roll = await d20Roll(rollData);
+  async rollAbilityTest(config={}, dialog={}, message={}) {
```

Comment:
Maybe we should taking this opportunity to deprecate `rollAbilityTest` in favour of `rollAbilityCheck`, so that it better matches the game term. Similarly with `rollAbilitySave` -> `rollSavingThrow`, with the assumption that we will eventually re-work the pipeline to match the game rules, i.e.:

```
rollSkillCheck -> rollAbilityCheck -> rollD20Test
rollDeathSave/rollConcentration -> rollSavingThrow -> rollD20Test
```

Then we could rename the private `#rollAbility` to `#rollAbilityTest` in order to align with the 'D20 Test' term, without it being confused with the public `rollAbilityTest` (after the deprecation period that is).

Maybe it's too early for that though. I just wondered what your thoughts were.

Reply:
Yeah, we can do that. I'll name the private method just `#rollD20Test`, since it is designed to work even without an ability specified.

Reply:
Yeah that sounds good. Attack rolls are also D20 Tests though, so we should use that name only so long as we plan that attack rolls will *eventually* go through this method and/or fire a `rollD20Test` hook some other way.

Reply:
They should definitely call the hook, but they might not be able to use the same method due to some of the roll preparation complications. I'll see when I get there.